fix custom mime type in out-display

eosfor opened the pull request:
looks like the `CustomMimeType` parameter was not in use at all. This way we can start using it and thus leverage a set of renderers like the one for vega-light. Example is also included in this pull request

eosfor commented:
@dotnet-policy-service agree

eosfor commented:
this is a very first attempt to send a pull request :D

@colombod, @jonsequitur, pls have a look

jonsequitur commented:
I was unaware of this property. I think it would be clearer to remove it as well as any constraints on the value of the `MimeType` property, which would be more consistent with how MIME types are used elsewhere.

jonsequitur commented:
Thanks for the PR! 

I have a couple of requests if you'd like to pursue it. 

* We've largely moved away from trying to document every feature in example notebooks. The usage example is helpful but doesn't need to part of this PR. 

* However, there should be an accompanying test which can serve as an example of expected usage as well as ongoing verification that it's working as expected.

Thanks!

eosfor commented:
@jonsequitur hopefully i did it right :). Removed the example from the PR and added a test i could come up with. 

But, actually, c# test won't look like a powershell example for many, i guss

jonsequitur commented on src/Microsoft.DotNet.Interactive.PowerShell/Commands/OutDisplayCommand.cs:
```diff
@@ -53,8 +55,17 @@ public sealed class OutDisplayCommand : PSCmdlet
+            displayedValue = display(obj, MimeType);
+        }
+
+        if (ParameterSetName == "CustomMimeType"){
```
If we completely remove the `CustomMimeType` property and set `MimeType` in the test (or in the notebook for that matter), does it work as expected?
